app/src/main/java/com/github/mik629/android/fundamentals/data/db/daos/MovieDao.kt

```diff
@@ -0,0 +1,48 @@
+package com.github.mik629.android.fundamentals.data.db.daos
+
+import androidx.room.*
+import com.github.mik629.android.fundamentals.data.db.models.*
+import com.github.mik629.android.fundamentals.data.db.models.MovieDbEntity.Companion.COLUMN_NAME_RATING
+import com.github.mik629.android.fundamentals.data.db.models.MovieDbEntity.Companion.MOVIES_TABLE_NAME
+
+@Dao
+interface MovieDao {
+    @Transaction
+    @Query(value = "SELECT * FROM $MOVIES_TABLE_NAME ORDER BY $COLUMN_NAME_RATING DESC")
+    suspend fun getAllMovies(): List<MovieWithActorsAndGenres>
```

Comment:
How? It's an interface. Tried converting it into abstract class but abstract method can not be marked with @Transaction annotation

Reply:
yes 
abstract class
it works with @Transaction

Reply:
`/Users/maxifier/AndroidStudioProjects/AndroidFundamentalsApp/app/build/tmp/kapt3/stubs/debug/com/github/mik629/android/fundamentals/data/db/daos/MovieDao.java:52: error: Method annotated with @Transaction must not be private, final, or abstract. It can be abstract only if the method is also annotated with @Query.`